[WIP] Add developer documentation. (#53)
hyperledger-labs/perun-doc: source/node/design_docs/interaction_alice_node_sdk.rst

  1. In case of collaborative close, the channel state will be directly
     concluded on the blockchain, without needing to register and wait for the
     challenge duration to expire.

  2. In case of non-collaborative close, the channel state will first have to
     be registered on the blockchain. Both the participants have to then wait
     for the challenge duration to pass and then conclude it. The challenge
     duration allows time for the other pariticipant to refuse if an older
     state was registered.
     
  Collaborative Close is an optimization to close the channel with lesser
  transactions on the blockchain (one instead of two) and with no waiting
  period.

Register Phase
--------------

In this phase, either a non-final state is registered or a final state is
directly concluded on the blockchain. This depends on if the pariticipant have
chosen non-collaborative or collaborative close respectively.

.. image:: ../../_generated/node/seq_3_register.svg
  :align: Center
  :alt: Image not available

If a non-final state was registered on the blockchain and other participant
sees that it is not the latest state, it can refute to the registered state by

> registering the latest state to the blockchain. This can go on until the no

Reviewer: ```suggestion
registering the latest state to the blockchain. This can go on until no
```

Author: igned-off-by: Manoranjith <ponraj.manoranjitha@in.bosch.com>

participants refute until challenge duration expires. Once it expires, the
channel progresses the settle phase.

If a final state was directly concluded on the blockchain, then also the
channel progresses to the settle phase.

Settle Phase
------------

In this phase, the node concludes the state of the channel on the blockchain
(only in case of non-collaborative close) and withdraws the funds back to the
user's account.

.. image:: ../../_generated/node/seq_4_settle.svg
  :align: Center
  :alt: Image not available

Once a channel is settled, user receives a channel close notification. Now the

> amount held by user in the concluded state of the channel will have been

Reviewer: ```suggestion
amount held by user in the concluded state of the channel can be 
```
